src/Features/Core/Portable/RemoveAsyncModifier/AbstractRemoveAsyncModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,261 @@
+
+            var diagnostic = context.Diagnostics.First();
+            var token = diagnostic.Location.FindToken(cancellationToken);
+            var node = token.GetAncestor(IsAsyncSupportingFunctionSyntax);
+            if (node == null)
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var methodSymbol = GetMethodSymbol(node, semanticModel, cancellationToken);
+
+            if (methodSymbol == null)
+            {
+                return;
+            }
+
+            if (ShouldOfferFix(methodSymbol.ReturnType, knownTypes))
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(c => FixAsync(document, diagnostic, c)),
+                    context.Diagnostics);
+            }
+        }
+
+        protected sealed override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var generator = editor.Generator;
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var compilation = semanticModel.Compilation;
+            var knownTypes = new KnownTypes(compilation);
+
+            foreach (var diagnostic in diagnostics.OrderByDescending(d => d.Location.SourceSpan.Start))
+            {
+                var token = diagnostic.Location.FindToken(cancellationToken);
+                var node = token.GetAncestor(IsAsyncSupportingFunctionSyntax);
+                if (node == null)
+                {
+                    Debug.Fail("We should always be able to find the node from the diagnostic.");
+                    continue;
+                }
+
+                var methodSymbol = GetMethodSymbol(node, semanticModel, cancellationToken);
+                if (methodSymbol == null)
+                {
+                    Debug.Fail("We should always be able to find the method symbol for the diagnostic.");
+                    continue;
+                }
+
+                // We might need to perform control flow analysis as part of the fix, so we need to do it on the original node
+                // so do it up front. Nothing in the fixer changes the reachability of the end of the method so this is safe
+                var controlFlow = GetControlFlowAnalysis(generator, semanticModel, node);
+                // If control flow couldn't be computed then its probably an empty block, which means we need to add a return anyway
+                var needsReturnStatementAdded = controlFlow == null || controlFlow.EndPointIsReachable;
+
+                editor.ReplaceNode(node, (updatedNode, generator) => RemoveAsyncModifier(generator, updatedNode, methodSymbol.ReturnType, knownTypes, needsReturnStatementAdded));
```

Comment:
```suggestion
                editor.ReplaceNode(node,
                    (updatedNode, generator) => RemoveAsyncModifier(generator, updatedNode, methodSymbol.ReturnType, knownTypes, needsReturnStatementAdded));
```

Nit: my preference is to be visible on a github screen width without horizontal scrolling.